networkx/classes/graph.py

```diff
@@ -1539,20 +1541,23 @@ def selfloop_edges(self, data=False, default=None):
         >>> G = nx.Graph()   # or DiGraph, MultiGraph, MultiDiGraph, etc
         >>> G.add_edge(1,1)
         >>> G.add_edge(1,2)
-        >>> G.selfloop_edges()
+        >>> list(G.selfloop_edges())
         [(1, 1)]
-        >>> G.selfloop_edges(data=True)
+        >>> list(G.selfloop_edges(data=True))
         [(1, 1, {})]
         """
         if data is True:
-            return [(n, n, nbrs[n])
-                    for n, nbrs in self.adj.items() if n in nbrs]
+            for n, nbrs in self.adj.items():
```

Comment:
I was trying to make this method return an iterator instead of list and `yield` will not work 2/3 tuple with a for loop. So, I made this change.